torch/linalg/__init__.py

```diff
@@ -248,42 +268,60 @@
 eig = _add_docstr(_linalg.linalg_eig, r"""
 linalg.eig(input, *, out=None) -> (Tensor, Tensor)
 
-Computes the eigenvalues and eigenvectors of a square
-matrix :attr:`input`, or of each such matrix in a batched :attr:`input`.
+Computes the eigenvalue decomposition of a square matrix or batch of square matrices.
+
+For a square matrix :math:`A`, this is defined as
 
-For a single matrix :attr:`input`, the tensor of eigenvalues `w` and the tensor of eigenvectors
-`V` decompose the :attr:`input` such that `input = V diag(w) V⁻¹`, where `V⁻¹` is the inverse of `V`.
+.. math::
+
+    A = V \operatorname{diag}(L) V^{-1}
+
+This decomposition exists if :math:`A` is diagonalizable (for example, when all its eigenvalues are different).
 
-The eigenvalues are not necessarily sorted.
-For real-valued input the resulting eigenvalues and eigenvectors will always be of complex type.
+The returned decomposition is represented as a namedtuple `(eigenvalues, eigenvectors)`,
+which corresponds to :math:`L`, :math:`V` above. If :attr:`input` is a batch of matrices,
+then `L`, `V` are also batched with the same batch dimensions.
 
-Supports input of float, double, cfloat and cdouble dtypes.
+Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
+The eigenvalues and eigenvectors will always be complex-valued, even if :attr:`input` is real.
 
-.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+.. note:: When :attr:`input` is on a CUDA device, this function synchronizes that device with the CPU.
 
-.. note:: The eigenvalues and eigenvectors are computed using LAPACK's and MAGMA's `geev` routine.
+.. note:: This function is computed using LAPACK's and MAGMA's `geev` routine for
```

Comment:
"routine" -> ""

If we want to consistently specify what these names refer to then we should consistently say "function."